dhis-2/dhis-api/src/main/java/org/hisp/dhis/i18n/I18nFormat.java

```diff
@@ -220,7 +221,18 @@ public String formatPeriod( Period period )
             return null;
         }
 
-        String typeName = period.getPeriodType().getName();
+        String typeName = period.getPeriodType().getName();        
+
+        if ( PeriodType.getCalendar().name().equals( "persian" ) )

```

Comment:
Please remove the hard-coding of this specific calendar in I18nFormat. We would like I18nFormat to be unaware of any period types. This should be built into the Calendar solution.